### PR TITLE
Added documentation section

### LINKED ISSUE #20: Update README to include Doxygen documentation
## Description
In previous changes, the code documentation was added to the repository, but the README file was not updated.

## DoD
- [x] Add the Documentation section in the README file

### PR DESCRIPTION
**Pull request by Damian-Mangold**
| Issue |
|---|
|Closes #20 |

## Description
The purpose of this PR is to update the README file by adding the documentation section.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,4 +28,13 @@ Please see the CMake documentation and CMakeLists.txt for more advanced usage.
 
 4. **Diagrams** All changes must be accompanied by a new (or changed) architectural diagrams.
 
+### Documentation
+
+Apart from the markdown that we can find in the readme of the project subfolders, the code has doxygen headers, which generates a technical documentation which gives some context of what the call tree is like, and what parameters and returns they have.
```
Comment (pereyra-m) [inline README.md:33]: The sentence ` ... which generates a technical documentation which gives some context of what the call tree is like ...` could be 

```
... which generate technical documentation that gives some context of what the call tree is like ...
```

Comment (Damian-Mangold) [reply README.md:33]: I copied the text from this [README](https://github.com/wazuh/wazuh-content/blob/main/README.md), maybe we should update it too?

Comment (pereyra-m) [reply README.md:33]: Maybe we should.

This is the result from https://app.grammarly.com/
grammarly

Comment (Damian-Mangold) [reply README.md]: done

**Review by pereyra-m — APPROVED** [review]

**Review by Dwordcito — APPROVED** [review]

**Pull request merged**